### PR TITLE
[Delivers #82735468] Barcode Rapid Entry

### PR DESCRIPTION
**Pull request by payten**
- refactor both the ILS a touch to create a consistent workflow
- barcode rapid entry form lists all selected display strings along with the existing barcode value.  The input for for each is empty to aid barcode-gun entry. The 'return' key in these fields will tab to the next to aid "rapidness". 

I couldn't quickly think of a quick way to perform the updates in the backend.. so this all could definitely use a review.

### CODE REVIEW

File: backend/controllers/container.rb [hunk 1]
```diff
@@ -92,4 +92,16 @@ class ArchivesSpaceService < Sinatra::Base
     json_response(result)
   end
 
+
+  Endpoint.post('/repositories/:repo_id/top_containers/bulk/barcodes')
+  .description("Bulk update barcodes")
+  .params(["barcode_data", String, "JSON string containing barcode data {uri=>barcode}", :body => true],
```
Comment (payten) [inline backend/controllers/container.rb]: Without creating a JSONModel, I post the barcode data as a JSON string  { uri => barcode_value, uri_2 => barocde_value_2 }.  I know it's a little ugly.. so any ideas would be great!

**Comment by payten** [comment]
Okidokes.. changes made after review.  Bulk update of barcode now ensures the barcodes are valid and this exception is displayed in the frontend nicely.

Also minor refactor of bulk delete to make consistent with the prototype-style of defining actions.

**Comment by jambun** [comment]
@payten i'll take a look at this tomorrow with mark doesn't get to it first.

**Comment by jambun** [comment]
Hi @payten - when I entered duplicate barcodes I got this in the flash:
translation missing: en.top_container.db_error - Database integrity constraint conflict: Java::JavaSql::SQLIntegrityConstraintViolationException: The statement was aborted because it would have caused a duplicate key value in a unique or primary key constraint or unique index identified by 'YALE_UNIQ_BARCODE' defined on 'TOP_CONTAINER'.

**Comment by marktriggs** [comment]
Ah, good catch.  Another case to test would be to take two existing top containers (with barcodes) and try swapping their barcodes.  I have a feeling that might throw a unique constraint, even though it should be OK.

**Comment by jambun** [comment]
right - that does indeed throw an error. in my example the error should be thrown, i was concerns by the ugliness of its presentation ;)

**Comment by jambun** [comment]
this branch has some failures on backend:test - but they don't originate from this work. i have a fix for the ones failing a call to object_graph in the calculate restricted pr ... i see another one about pagination which looks completely unrelated to our work - i'll take a quick look.

**Comment by jambun** [comment]
actually that pagination failure (is not actually about pagination) looks like a kooky side effect of the failed call to object_graph problem. sorry for the noise.

**Comment by jambun** [comment]
i guess it's a tough call - and might require feedback from the yale folks - but maybe it is better to have the existing barcode in the field. it gets selected for easy replacement when tabbing or return-keying into the field.

**Comment by marktriggs** [comment]
Agreed w/ the presentation thing, yeah.  The swapping case was just a random additional thought :)  In other places I've NULLed out the column prior to setting the new value, which tricks MySQL into doing the right thing.  That approach should be fine here too.

**Comment by payten** [comment]
Ah yep.. looks like we're missing an attribute-error mapping for the duplicate barcode sequel exception. 

Yeah, I originally had the barcode in the field, but removed it just in case it conflicted with the barcode-scanner entry.  Without the values (instead in the column next to the input), it also makes it easier to remove the barcode for a set of items... which might be a positive or negative!  Easy change.. just need to know which way to go.

And yep.. I'll add a test for the swapping barcodes issue.  I think nulling out all the barcodes first is the go.

Cheers! Will try to get onto this Monday/Tuesday...

**Comment by payten** [comment]
@jambun, @marktriggs - I've updated the PR to fix the error messaging and barcode-swapping issue.

I've left the input on the rapid entry form empty for the moment until we confirm what would better suit their barcode scanners.  Note: 'return' in the input field will focus the next input, which should assist when using a scanner.

**Comment by jambun** [comment]
:peach:

**Pull request merged**